File: Numeros Pseudo-Aleatorios/XorShift 32, 64 e 128 bits/128 bits/xorshift128.dart
/*
 PRNG .......: XorShift (128 bits)
 Ano ........: 2003
 Autor ......: George Marsaglia
*/

class XorShift128State {
  int seed = 0;
}

int xorShift128(XorShift128State state) {
	
  int random = state.seed;
  random ^= random << 13;
  random ^= random >> 7;
  random ^= random << 17;
  state.seed = random;
  return random & 0x7FFFFFFFFFFFFFFF;
}

void main() {
  var Xn1 = XorShift128State();
  var Xn2 = XorShift128State();

  Xn1.seed = 1234;
  Xn2.seed = DateTime.now().millisecondsSinceEpoch;

  print('GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)');
  print('PSEUDORANDOM NUMBER GENERATOR (PRNG)');
  print('PADRAO: XorShift (128 bits)');
  print('=================================================');
  print('');
  print('   Semente: 1234          Semente: TIME');
  print('   --------------------   --------------------');

  for (int i = 1; i <= 10; i++) {
    print('${xorShift128(Xn1).toString().padLeft(23)}${xorShift128(Xn2).toString().padLeft(23)}');
  }
  
}

File: Numeros Pseudo-Aleatorios/XorShift 32, 64 e 128 bits/128 bits/xorshift128.go
/*
 PRNG .......: XorShift (128 bits)
 Ano ........: 2003
 Autor ......: George Marsaglia
*/

package main

import (
	"fmt"
	"time"
)

type xorShift128State struct {
	seed uint64
}

func xorshift128(state *xorShift128State) uint64 {
	random := state.seed
	random ^= random << 13
	random ^= random >> 7
	random ^= random << 17
	state.seed = random
	return random & 0x7FFFFFFFFFFFFFFF
}

func main() {
	var Xn1 xorShift128State
	var Xn2 xorShift128State

	Xn1.seed = 1234
	Xn2.seed = uint64(time.Now().UnixNano())

	fmt.Println("GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)")
	fmt.Println("PSEUDORANDOM NUMBER GENERATOR (PRNG)")
	fmt.Println("PADRAO: XorShift (128 bits)")
	fmt.Println("=================================================")
	fmt.Println("")
	fmt.Println("   Semente: 1234          Semente: TIME")
	fmt.Println("   --------------------   --------------------")

	for i := 1; i <= 10; i++ {
		fmt.Printf("%23d%23d\n", xorshift128(&Xn1), xorshift128(&Xn2))
	}
}
